File: game-server/src/main/java/com/kingston/mmorpg/game/scene/model/WorldMap.java
package com.kingston.mmorpg.game.scene.model;

import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;
import java.util.concurrent.locks.ReentrantReadWriteLock;

/**
 * @author kingston 世界地图模型
 */
public class WorldMap {

	/** 以该地图为模型的所有场景 */
	private ConcurrentMap<Long, Scene> scenes = new ConcurrentHashMap<>();

	/** 地图模型id */
	private int modelId;
	
	/** 读写锁 */
	private ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
	private ReentrantReadWriteLock.ReadLock readLock = lock.readLock();
	private ReentrantReadWriteLock.WriteLock writeLock = lock.writeLock();
	
	public void addScene(long id, Scene scene) {
		scenes.put(id, scene);
	}
	
	
}

File: game-server/src/main/java/org/forfun/mmorpg/game/scene/model/WorldMap.kt
package org.forfun.mmorpg.game.scene.model

import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.locks.ReentrantReadWriteLock

/**
 *  世界地图模型
 */
class WorldMap {

    /** 以该地图为模型的所有场景 */
    private val scenes = ConcurrentHashMap<Long, Scene>()

    /** 地图模型id */
    var modelId: Int = 0

    /** 读写锁 */
    private val lock = ReentrantReadWriteLock()
    private val readLock = lock.readLock()
    private val writeLock = lock.writeLock()

    fun addScene(id: Long, scene: Scene) {
        scenes[id] = scene
    }
}
